crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -352,12 +352,16 @@ impl EventCache {
     ///   time to do so (new subscribers might have spawned in the meanwhile).
     #[instrument(skip_all)]
     async fn auto_shrink_linked_chunk_task(
-        inner: Arc<EventCacheInner>,
+        inner: Weak<EventCacheInner>,
         mut rx: mpsc::Receiver<AutoShrinkChannelPayload>,
     ) {
         while let Some(room_id) = rx.recv().await {
             trace!(for_room = %room_id, "received notification to shrink");
 
+            let Some(inner) = inner.upgrade() else {
+                return;
+            };
```

Comment:
Do you believe we can test the previous buggy behaviour?

Reply:
I think it would be quite important to get this honestly. Thoughts?

Reply:
I don't think so, but I won't be opposed to you writing a test :)